book/src/dev/rfcs/0010-v5-transaction.md

```diff
@@ -0,0 +1,262 @@
+
+To highlight changes most of the document comments from the code snippets in the [reference section](#reference-level-explanation) were removed.
+
+## Sapling Changes Overview
+[sapling-changes-overview]: #sapling-changes-overview
+
+V4 and V5 transactions both support sapling, but the underlying data structures are different. So need to make the sapling data types generic over the V4 and V5 structures.
+
+In V4, anchors are per-spend, but in V5, they are per-transaction.
+
+For consistency, also we move some fields into the `ShieldedData` type, and rename some fields and types.
+
+## Orchard Additions Overview
+[orchard-additions-overview]: #orchard-additions-overview
+
+V5 transactions are the only ones that will support orchard transactions with `Orchard` data types.
+
+Orchard uses `Halo2Proof`s with corresponding signature type changes. Each Orchard `Action` contains a spend and an output. Placeholder values are substituted for unused spends and outputs.
+
+## Other Transaction V5 Changes
+[other-transaction-v5-changes]: #other-transaction-v5-changes
+
+The order of some of the fields changed from V4 to V5. For example the `lock_time` and `expiry_height` were moved above the transparent inputs and outputs.
+
+Zebra enums and structs put fields in serialized order. Composite fields are ordered based on **last** data deserialized for each field.
+
+# Reference-level explanation
+[reference-level-explanation]: #reference-level-explanation
+
+## Sapling Changes
+[sapling-changes]: #sapling-changes
+
+We know by protocol (2nd table of [Transaction Encoding and Consensus](https://zips.z.cash/protocol/nu5.pdf#txnencodingandconsensus)) that V5 transactions will support sapling data however we also know by protocol that spends ([Spend Description Encoding and Consensus](https://zips.z.cash/protocol/nu5.pdf#spendencodingandconsensus), See †) and outputs ([Output Description Encoding and Consensus](https://zips.z.cash/protocol/nu5.pdf#outputencodingandconsensus), See †) fields change from V4 to V5.
+
+### Changes to V4 Transactions
+[changes-to-v4-transactions]: #changes-to-v4-transactions
+
+Here we have the proposed changes for V4 transactions:
+* make `sapling_shielded_data` use the `PerSpendAnchor` anchor variant
+* rename `shielded_data` to `sapling_shielded_data`
+* move `value_balance` into the `sapling::ShieldedData` type
+* order fields based on the **last** data deserialized for each field
+
+```rust
+enum Transaction::V4 {
+    inputs: Vec<transparent::Input>,
+    outputs: Vec<transparent::Output>,
+    lock_time: LockTime,
+    expiry_height: block::Height,
+    joinsplit_data: Option<JoinSplitData<Groth16Proof>>,
+    sapling_shielded_data: Option<sapling::ShieldedData<PerSpendAnchor>>,
+}
+```
+
+### Anchor Variants
+[anchor-variants]: #anchor-variants
+
+We implement an `AnchorVariant` generic type trait, because V4 transactions have a per-`Spend` anchor, but V5 transactions have a shared anchor:
+
```

Comment:
We can specify in what file we will add the `PerSpendAnchor`, most probably `zebra-chain/src/transaction/shielded_data.rs`

Reply:
We should move this sentence as well:
> As this is Sapling specific we propose to move this file to `zebra-chain/src/sapling/shielded_data.rs`.

Reply:
> We can specify in what file we will add the `PerSpendAnchor`, most probably `zebra-chain/src/transaction/shielded_data.rs`

Sorry, i meant `zebra-chain/src/sapling/shielded_data.rs`. I think the code for `AnchorVariant` will probably need to be there.

Reply:
I added a specific file for every new type in 4cba267e98b48cce9e1d23760b25db282e517935